Dockerfile

```diff
@@ -4,92 +4,81 @@ FROM debian:latest
-RUN echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list
-RUN apt-get -y update \
+RUN >&2 echo "# Install PHP 7 with extensions and update settings" \
+    && wget -O - https://www.dotdeb.org/dotdeb.gpg | apt-key add - \
+    && echo "deb https://packages.dotdeb.org $(lsb_release -sc) all" | tee /etc/apt/sources.list.d/php.list \
+    && echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list \
+    && apt-get -y update \
     && apt-get install -y php7.0-fpm php7.0-cli php7.0-mbstring php7.0-mcrypt php7.0-intl \
-    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl
-# Update some PHP settings
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini
-# Composer updates can require a lot of memory
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini
-RUN phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl
+    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl php-imagick \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini \
+    && phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl imagick
 
 # Copy in php-fpm config file
 COPY php-fpm.conf /etc/php/7.0/fpm/php-fpm.conf
 
-# Install composer
-RUN wget -O - https://getcomposer.org/installer | php \
+RUN >&2 echo "# Installing tooling (composer, nodejs)..." \
+    && EXPECTED_SIGNATURE=$(wget -q -O - 'https://composer.github.io/installer.sig') \
+    && wget -O composer-installer.php 'https://getcomposer.org/installer' \
+    && INSTALLER_SIGNATURE=$(openssl dgst --sha384 -hex composer-installer.php | sed 's/^.* \(.*\)$/\1/') \
+    && if [ "$INSTALLER_SIGNATURE" != "$EXPECTED_SIGNATURE" ] ; then >&2 echo 'ERROR: Invalid installer signature' && exit 1 ; fi \
+    && php composer-installer.php \
     && mv composer.phar /usr/local/bin/composer \
-    && composer self-update
-
-# Install nodejs and npm
-RUN curl -sL https://deb.nodesource.com/setup_6.x | bash - \
-    && apt-get install -y nodejs
-
-# Install nginx and supervisord from dotdeb
-RUN apt-get -y update && apt-get -y install supervisor nginx
-
-# One last update for everything
-RUN apt-get -y update && apt-get -y dist-upgrade
+    && rm composer-installer.php \
+    && curl -sL https://deb.nodesource.com/setup_6.x | bash -
```

Comment:
Maybe should be before `apt-get install nodejs`. Needs investigation.

Reply:
Done.